chainbase/src/main/java/org/tron/core/service/RewardViCalService.java

```diff
@@ -75,8 +74,14 @@ public RewardViCalService(@Autowired  DynamicPropertiesStore propertiesStore,
     this.witnessStore = witnessStore.getDb();
   }
 
-  @PostConstruct
-  private void init() {
+  public void init() {
+    // after init, we can get the latest block header number from db
+    this.newRewardCalStartCycle = this.getNewRewardAlgorithmEffectiveCycle();
+    boolean ret = this.newRewardCalStartCycle != Long.MAX_VALUE;
+    if (ret) {
+      // checkpoint is flushed to db, we can start rewardViCalService immediately
+      lastBlockNumber = Long.MAX_VALUE;
+    }
```

Comment:
There is no need to define this variable `ret`.
The style needs to be consistent,  `lastBlockNumber` -> `this.lastBlockNumber`